src/main/java/com/minecrafttas/common/events/client/EventCamera.java

```diff
@@ -0,0 +1,47 @@
+package com.minecrafttas.common.events.client;
+
+import com.minecrafttas.common.events.EventBase;
+import com.minecrafttas.common.events.EventListener;
+
+public interface EventCamera extends EventBase{
+	
+	public CameraData onCameraEvent(CameraData dataIn);
+	
+	public static CameraData fireCameraEvent(CameraData dataIn) {
+		for (EventBase eventListener : EventListener.getEventListeners()) {
+			if(eventListener instanceof EventCamera) {
+				EventCamera event = (EventCamera) eventListener;
+				CameraData data = event.onCameraEvent(dataIn);
+				if(data.equals(dataIn)) {
+					return data;
+				}
+			}
+		}
+		return dataIn;
+	}
```

Comment:
are you sure you want this in the interface?

Reply:
Yes